src/Api/AdminConsole/Authorization/Requirements/ManageUsersRequirement.cs

```diff
@@ -0,0 +1,17 @@
+﻿#nullable enable
+
+using Bit.Core.Context;
+using Bit.Core.Enums;
+
+namespace Bit.Api.AdminConsole.Authorization.Requirements;
+
+public class ManageUsersRequirement : IOrganizationRequirement
+{
+    public async Task<bool> AuthorizeAsync(
+        CurrentContextOrganization? organizationClaims,
+        Func<Task<bool>> isProviderUserForOrg)
+        => organizationClaims is
+    { Type: OrganizationUserType.Owner or OrganizationUserType.Admin } or
+    { Permissions.ManageUsers: true }
+            || await isProviderUserForOrg();
```

Comment:
🎨 This code is difficult to read. Consider using a switch expression.

```suggestion
public async Task<bool> AuthorizeAsync(
    CurrentContextOrganization? organizationClaims,
    Func<Task<bool>> isProviderUserForOrg)
    => organizationClaims switch
    {
        { Type: OrganizationUserType.Owner } => true,
        { Type: OrganizationUserType.Admin } => true,
        { Permissions.ManageUsers: true } => true,
        _ => await isProviderUserForOrg()
    };
```

Reply:
Nice suggestion 🎨